compiler/plugins/target/WebGPU/CMakeLists.txt

```diff
@@ -4,18 +4,20 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-if(NOT IREE_TARGET_BACKEND_WEBGPU)
-  return()
-endif()
-
 iree_add_all_subdirs()
 
+iree_compiler_register_plugin(
+  PLUGIN_ID
+    hal_target_webgpu
+  TARGET
+    ::WebGPU
+)
+
 iree_cc_library(
   NAME
     WebGPU
```

Comment:
We can also move the supporting build system configuration into this plugin directory, similar to how Torch does this: https://github.com/openxla/iree/blob/main/compiler/plugins/input/Torch/torch-mlir/CMakeLists.txt

These are the pieces that can move:

* https://github.com/openxla/iree/blob/main/build_tools/third_party/tint/CMakeLists.txt
* https://github.com/openxla/iree/blob/ac9548469e8e46f8ad6ea29c1c7144a774ebba19/CMakeLists.txt#L930-L946